src/store/dtree.ts

```diff
@@ -114,7 +114,44 @@ class DtreeStore {
   actionHistory: IDtreeSetArguments[] = []
   actionHistoryIndex = -1
 
+  readonly dtreeSet = new DtreeSetAsyncStore()
+  readonly dtreeCounts = new DtreeCountsAsyncStore()
+
+  get pointCounts() {
+    const counts = this.isXl
+      ? this.dtreeCounts.data?.['point-counts']
+      : this.dtreeSetData?.['point-counts']
+
+    return counts ?? []
+  }
+
   constructor() {
+    reaction(
+      () => this.dtreeSetData,
+      response => {
+        if (response) {
+          stepStore.setSteps(adaptDtreeSetToSteps(response))
```

Comment:
adapters are not utility functions. They are supposed to modify the outcome data interface. They work like this: Interface A --> adapter --> Interface B. `getDtreeSet` should return another interface, which would include `steps` inside of it. In other words, you must apply this adapter in `getDtreeSet` method and add `steps` to the response. And in this place you're going to call just `this.dtreeSetData?.steps`.